x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityStatusChangeListener.java

```diff
@@ -40,6 +41,11 @@ public synchronized void licenseStateChanged() {
         // old state might be null (undefined) so do Object comparison
         if (Objects.equals(newState, securityEnabled) == false) {
             logger.info("Active license is now [{}]; Security is {}", licenseState.getOperationMode(), newState ? "enabled" : "disabled");
+            if (newState == false) {
```

Comment:
I think this is fine. The only possibility that I can think of for a running cluster to have securityEnabled changing from `true` to `false` is something like: **Intially** security is implicitly enabled, i.e. `xpack.security.enabled` is not configured and the cluster has a Gold+ license. **Later** the license is either downgraded or deleted or a new trial is started which causes the security to be implicitly disabled. In another word, the warning should only be shown when the cluster is reverted back to the "security is implicitly disabled" state.

Reply:
Right, and it says something similar in a comment above.

Reply:
Yang's commented isn't quite right - this message will be printed when the state changed from `null` to `false` as well, so it will be printed on startup for any cluster with security implicitly disabled even if the license doesn't change.
Since the listener isn't registered when security is explicitly disabled, it won't print out for those cases, but it's not strictly confined to the `true` -> `false` cases.